packages/eslint-plugin/src/rules/shorthand-property-sorting/index.ts

```diff
@@ -0,0 +1,124 @@
+import type { Rule } from 'eslint';
+
+import { shorthandFor } from './utils';
+
+const findProgramRoot = (node: Rule.Node) => {
+  const parent = node.parent;
+  if (!parent) {
+    return null;
+  }
+  if (parent.type === 'Program') {
+    return parent;
+  }
+  return findProgramRoot(parent);
+};
+
+export const shorthandFirst: Rule.RuleModule = {
+  meta: {
+    docs: {
+      recommended: true,
+      description:
+        'When using both shorthand and longhand properties, compiled scrambles the properties and it is unintuative to understand which properties override which others. Ordering the shorthand first ensures that longhand overrides shorthand.',
+      url: 'https://github.com/atlassian-labs/compiled/tree/master/packages/eslint-plugin/src/rules/shorthand-property-sorting',
+    },
+    messages: {
+      'shorthand-first':
+        'When using both shorthand and longhand properties, the shorthand property should be first.',
+    },
+    type: 'problem',
```

Comment:
if we can't preserve comments, maybe safer to change this rule from a `problem` to a `suggestion`

see here https://eslint.org/docs/latest/extend/custom-rules#providing-suggestions for what you'd need to change